### PR TITLE
upgraded kcl and aws java sdk versions

### PR DESCRIPTION
**Pull request by AvinashChowdary**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <artifactId>amazon-kinesis-client</artifactId>
   <packaging>jar</packaging>
   <name>Amazon Kinesis Client Library for Java</name>
-  <version>1.14.4-SNAPSHOT</version>
+  <version>1.14.5</version>
```
Comment (jonatious) [inline pom.xml]: Change log says version 1.14.4?

**Comment by jonatious** [comment]
Can you squash the commits into one, so it is easier to review the final changes?

File: pom.xml (hunk 1 above)
Comment (AvinashChowdary) [reply pom.xml]: Changed in [this](https://github.com/awslabs/amazon-kinesis-client/pull/821/commits/27b5df15315c15d9a6bb090abb1d39f6344df8c9) commit

**Comment by AvinashChowdary** [comment]
Squashed into one commit.

File: README.md [hunk 2]
```diff
@@ -34,7 +34,10 @@ To make it easier for developers to write record processors in other languages,
 
 ## Release Notes
 
-### Latest Release (1.14.3 - May 3, 2021)
+### Latest Release (1.14.4 - June 11, 2021)
+* [#816](https://github.com/awslabs/amazon-kinesis-client/pull/816) Updated the Worker shutdown logic to make sure that the `LeaseCleanupManager` also terminates all the threads that it has started.
```
Comment (joshua-kim) [inline README.md:38]: Did we get approval for these release notes from doc writer?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 
-### Latest Release (1.14.3 - May 3, 2021)
+### Latest Release (1.14.4 - June 11, 2021)
```
Comment (joshua-kim) [inline CHANGELOG.md:3]: We should link the related milestone for v.1.14.4 in these release notes (https://github.com/awslabs/amazon-kinesis-client/milestone/61)), and also tag any PR's (#816) being merged as part of this release under this milestone.

**Review by joshua-kim — CHANGES_REQUESTED** [review]
Let's add the milestone for the release. Did the doc writer approve these release notes?

File: README.md (hunk 2 above)
Comment (AvinashChowdary) [reply README.md:39]: Yes.

File: CHANGELOG.md (hunk 3 above)
Comment (AvinashChowdary) [reply CHANGELOG.md:3]: Done

File: pom.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
   </licenses>
 
   <properties>
-    <aws-java-sdk.version>1.11.844</aws-java-sdk.version>
+    <aws-java-sdk.version>1.12.3</aws-java-sdk.version>
```
Comment (joshua-kim) [inline pom.xml:28]: Can you make this diff a separate PR? This change is independent of the release, so we should include the AWS SDK upgrade PR in the release notes for any customers who might need to override their SDK. [Example previous PR](https://github.com/awslabs/amazon-kinesis-client/commit/1d5099cef4459f2f1a4804d5daa2ccf0352f3a6d), [example previous release notes](https://github.com/awslabs/amazon-kinesis-client/commit/9db3a11ffe04bdac412732afa408bccc8629183b).

**Review by joshua-kim — CHANGES_REQUESTED** [review]
One last comment to break up this PR

File: pom.xml (hunk 4 above)
Comment (AvinashChowdary) [reply pom.xml:28]: Updated this PR to contain only SDK version bump.

**Review by joshua-kim — APPROVED** [review]
LGTM

File: pom.xml (hunk 4 above)
Comment (joshua-kim) [reply pom.xml:28]: Thanks!

**Pull request merged**